src/gwt/acesupport/acemode/stan_highlight_rules.js

```diff
@@ -1,134 +1,186 @@
+    // ---
 
     this.$rules = {
-	"start" : [
-	    {
-		token : "comment",
-		regex : "\\/\\/.*$"
-	    }, {
-		token : "comment",
-		regex : "#.*$"
-	    }, {
-		token : "comment", // multi line comment
-		merge : true,
-		regex : "\\/\\*",
-		next : "comment"
-	    }, {
-		token : "keyword.blockid",
-		regex : "functions|data|transformed\\s+data|parameters|" +
-			 "transformed\\s+parameters|model|generated\\s+quantities"
-	    }, {
-		token : "string", // single line
-		regex : '["][^"]*["]'
-	    }, {
-		token : "constant.numeric",
-		regex : "[+-]?\\d+(?:(?:\\.\\d*)?(?:[eE][+-]?\\d+)?)?\\b"
-	    }, {
-		token : "keyword.operator", // truncation
-		regex : "\\bT(?=\\s*\\[)"
-	    }, {
-		// highlights everything that looks like a function
-		token : function(value) {
-		    if (keywordFunctionsMap.hasOwnProperty(value)) {
-			return "keyword"
-		    } else {
-			return "support.function"
-		    }
-		},
-		regex : variableName + "(?=\\s*\\()"
-	    }, {
-		token : keywordMapper,
-		regex : variableName + "\\b"
-	    }, {
-		token : "keyword.operator",
-		regex : "<-|~"
-	    }, {
-		token : "keyword.operator",
-		regex : "\\|\\||&&|==|!=|<=?|>=?|\\+|-|\\.?\\*|\\.?/|\\\\|\\^|!|'|%"
-	    }, {
-		token : "punctuation.operator",
-		regex : ":|,|;|="
-	    }, {
-		token : "paren.lparen.keyword.operator",
-		regex : "[\\[\\(\\{]"
-	    }, {
-		token : "paren.rparen.keyword.operator",
-		regex : "[\\]\\)\\}]"
-	    }, {
-		token : "text",
-		regex : "\\s+"
-	    }
-	],
-	"comment" : [
-	    {
-		token : "comment", // closing comment
-		regex : ".*?\\*\\/",
-		next : "start"
-	    }, {
-		token : "comment", // comment spanning whole line
-		merge : true,
-		regex : ".+"
-	    }
-	]
+      "start" : [
+        {
+          token : "comment.line.number-sign",
+          regex : "\\/\\/.*$"
+        },
+        {
+          token : "comment.line.double-dash",
+          regex : "#.*$"
+        },
+        {
+          token : "comment.block", // multi line comment
+          regex : "\\/\\*",
+          merge : true,
+          next : "comment"
+        },
+        {
+          // semantically this is closer to entity.name.type or entity.name.section
+          // however the highlighting does not look good for those.
+          token : "keyword.other.block",
+          regex : "\\b(functions|data|transformed\\s+data|parameters|transformed\\s+parameters|model|generated\\s+quantities)\\b"
+        },
+        {
+          // Stan is very strict on what characters it allows in strings
+          // only allows the space character and any visible ASCII character
+          // except the backslash \ and double quote " characters
+          // needs to be a single line
+          token : "string.quoted.double",
+          regex : '["][ a-zA-Z0-9~@#$%^&*_\'`\\-+={}[\\]()<>|/!?.,;:]*["]'
```

Comment:
There really are no strings in the Stan language. The `'` character is legal (but it means the transpose operation) while the `"` character would only appear in a comment (like `// this is a "comment"`) or in a preprocessor directive (like `#include "my_stuff.stan"`). The include statements are not even officially part of the Stan language yet, but `rstan::stan_builder` can preprocess them.